osu.Game/Screens/Backgrounds/BackgroundScreenBeatmap.cs

```diff
@@ -36,28 +36,34 @@ public virtual WorkingBeatmap Beatmap
 
                 beatmap = value;
 
-                Schedule(() =>
-                {
-                    LoadComponentAsync(new BeatmapBackground(beatmap), b => Schedule(() =>
-                    {
-                        float newDepth = 0;
-                        if (Background != null)
-                        {
-                            newDepth = Background.Depth + 1;
-                            Background.FinishTransforms();
-                            Background.FadeOut(250);
-                            Background.Expire();
-                        }
-
-                        b.Depth = newDepth;
-                        fadeContainer.Add(Background = b);
-                        Background.BlurSigma = BlurTarget;
-                        StoryboardReplacesBackground.BindTo(fadeContainer.StoryboardReplacesBackground);
-                    }));
-                });
+                Schedule(() => { LoadComponentAsync(new BeatmapBackground(beatmap), b => Schedule(() => backgroundLoaded(b))); });
```

Comment:
This'll cause a second `BeatmapBackground` load to still happen for the same beatmap in update (the first now being inside `load()`).

As discussed, probably want a `if (LoadState < LoadState.Ready) return;` here.